src/ads/RewardedAd.ts

```diff
@@ -158,4 +150,37 @@ export class RewardedAd extends MobileAd implements MobileAdInterface {
     }
     return this._googleMobileAds.native.rewardedShow(this._requestId, this._adUnitId, options);
   }
+
+  addAdEventsListener<T extends AdEventType | RewardedAdEventType>(
+    listener: AdEventsListener<T>,
+  ): () => void {
+    if (!isFunction(listener)) {
+      throw new Error("RewardedAd.addAdEventsListener(*) 'listener' expected a function.");
+    }
+
+    return this._addAdEventsListener(listener);
+  }
+
+  addAdEventListener<T extends AdEventType | RewardedAdEventType>(
+    type: T,
+    listener: AdEventListener<T>,
+  ) {
+    if (
+      !isOneOf(type, Object.values(AdEventType)) &&
+      !isOneOf(type, Object.values(RewardedAdEventType))
+    ) {
```

Comment:
As a slight refactor in the hypothetical change where the superclass has concrete implementations, if the superclass did an extract method refactor on type validation  such that `verifyAdEventListenerType: boolean` was a superclass method, then you could override just that to check for the two types here (vs the default AdEventType) and the rest of the functionality you want is de-duplicated via just a super call